### PR TITLE
Write excluded prefixes to configmap instead of file

### LINKED ISSUE #14: Write excluded prefixes to configmap instead of file
Now we need to deploy one excluded prefixes collector for each NS Manager. But we can write prefixes to configmap instead of file , like in monorepo, and thus we'll need only one prefixes collector for all NS Managers.

### PR DESCRIPTION
**Pull request by tigrulya-exe**
Should fix #14

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,45 +3,17 @@ module cmd-exclude-prefixes-k8s
...
 	github.com/sirupsen/logrus v1.6.0
 	github.com/stretchr/testify v1.6.1
 	go.uber.org/goleak v1.0.1-0.20200717213025-100c34bdc9d6
 	k8s.io/api v0.18.1
 	k8s.io/apimachinery v0.18.2
 	k8s.io/client-go v0.18.1
-	k8s.io/kubernetes v1.18.1
-)
-
-replace (
```
Comment (denis-tingajkin) [inline go.mod:22]: nice

File: internal/prefixcollector/collector.go [hunk 2]
```diff
@@ -47,43 +52,84 @@ type Notifier interface {
...
+				return
+			}
+
+			configMap, ok := event.Object.(*apiV1.ConfigMap)
+			if !ok || configMap.Name != epc.nsmConfigMapName {
+				continue
+			}
+
+			if event.Type == watch.Error {
+				logrus.Errorf("Error during nsm configmap %v/%v watch: %v", epc.nsmConfigMapNamespace, epc.nsmConfigMapName, err)
```
Comment (denis-tingajkin) [inline internal/prefixcollector/collector.go:116]: Let's use `log.Entry(ctx).WithFields` here

File: internal/prefixcollector/collector.go [hunk 3]
```diff
@@ -47,43 +52,84 @@ type Notifier interface {
...
+			if !ok {
+				return
+			}
+
+			configMap, ok := event.Object.(*apiV1.ConfigMap)
+			if !ok || configMap.Name != epc.nsmConfigMapName {
+				continue
+			}
+
+			if event.Type == watch.Error {
```
Comment (denis-tingajkin) [inline internal/prefixcollector/collector.go:115]: [minor] Can we use here switch by type?

File: main.go [hunk 4]
```diff
@@ -39,10 +39,11 @@ const (
...
 // Config - configuration for cmd-exclude-prefixes-k8s
 type Config struct {
-	ExcludedPrefixes   []string `desc:"List of excluded prefixes" split_words:"true"`
-	ConfigMapNamespace string   `default:"default" desc:"Namespace of kubernetes config map" split_words:"true"`
-	ConfigMapName      string   `default:"nsm-config" desc:"Name of kubernetes config map" split_words:"true"`
-	PrefixesFilePath   string   `default:"/var/lib/networkservicemesh/config/excluded_prefixes.yaml" desc:"Excluded prefixes file absolute path" split_words:"true"`
+	ExcludedPrefixes      []string `desc:"List of excluded prefixes" split_words:"true"`
+	ConfigMapNamespace    string   `default:"default" desc:"Namespace of user config map" split_words:"true"`
+	ConfigMapName         string   `default:"excluded-prefixes-config" desc:"Name of user config map" split_words:"true"`
+	NSMConfigMapNamespace string   `default:"default" desc:"Namespace of nsm config map" split_words:"true"`
```
Comment (denis-tingajkin) [inline main.go:45]: Can exclude prefix service find it's a namespace? In this case, we can remove this parameter.

**Review by denis-tingajkin — COMMENTED** [review]
I like these changes, added a few comments

File: internal/prefixcollector/collector.go (hunk 2 above)
Comment (tigrulya-exe) [reply internal/prefixcollector/collector.go]: Used

File: internal/prefixcollector/collector.go (hunk 3 above)
Comment (tigrulya-exe) [reply internal/prefixcollector/collector.go]: Replaced with switch

File: main.go (hunk 4 above)
Comment (tigrulya-exe) [reply main.go]: Now we get current namespace from mounted serviceaccount

File: main.go [hunk 5]
```diff
@@ -86,16 +88,25 @@ func main() {
...
 		prefixcollector.NewEnvPrefixSource(envPrefixes),
 		prefixcollector.NewKubeAdmPrefixSource(ctx, cond),
 		prefixcollector.NewKubernetesPrefixSource(ctx, cond),
 		prefixcollector.NewConfigMapPrefixSource(ctx, cond, config.ConfigMapName, config.ConfigMapNamespace),
 	)
 
+	if err != nil {
+		span.Logger().Fatalf("Error creating collector: %v", err)
+	}
+
```
Comment (denis-tingajkin) [inline main.go:109]: It looks like this block can be removed

Comment (tigrulya-exe) [reply main.go]: Removed

**Review by denis-tingajkin — APPROVED** [review]

**Pull request merged**